app/src/main/java/org/gnucash/android/model/Money.java

```diff
@@ -199,7 +199,7 @@ private void setCommodity(@NonNull Commodity commodity){
 	 * Sets the commodity for the Money
 	 * @param currencyCode ISO 4217 currency code
 	 */
-	private void setCommodity(@NonNull String currencyCode){
+	public void setCommodity(@NonNull String currencyCode){
```

Comment:
Was just going through the pr and this caught my eye. 
`Money` objects are meant to be immutable, just like Java Strings. Could you change this back to private and use `Money.withCurrency()` at the relevant places instead?